### PR TITLE
JS OOP

### PR DESCRIPTION
**Pull request by Shrai-dev**
### JS OOP (Frogger game)

[Demo](https://shrai-dev.github.io/frontend-nanodegree-arcade-game/)
[Code base](https://github.com/Shrai-dev/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@Shrai-dev you still refer to outdated submission instructions.
image

Not a big problem, just get used to a better git flow.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Shrai-dev good job.

Please check yourself against the checklist below and confirm your code base is changed to comply.

 * [ ] it is OK to employ ES6 features like `const`, `let` etc.
 * [ ] the code is very [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
 * [ ] OO is implemented using JS prototype chain object model (**not** ES6 classes syntax)
 * Requirements re **Constants**:
   * [ ] all numbers like block dimensions, initial locations are defined as named constants
         (e.g. `const STEP = 101;`) as otherwise numbers scattered across code base look cryptic;
         named constants add semantic meaning and improve readability
   * [ ] every number that have a semantic purpose (like those listed above) should be defined as constants;
         think of how your code **reads** - the closer to plain English the better
   * [ ] there are core constants and derived constants
         (e.g. derived constant `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants 
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`
 * Requirements re **OOP**:
   * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
         (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
         Hint: pass `Player` instance as an argument to every enemy
   * [ ] Separation of Concerns principle is followed
         (e.g. `update` method does only rendering and doesn't contain any unrelated **inline** code;
         for example collision check is defined as a dedicated method and only called from inside `update`)
   * [ ] Nice To Have: properties common for some classes are generalized into a base class
         (e.g. there is `Character` base class, which is extended by `Enemy` and `Player` classes)
   * [ ] class extension is implemented using `Subclass.prototype = Object.create(Superclass.prototype)`,
         not `Subclass.prototype = new Superclass(params);`;
         [Useful resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)

File: submissions/Shrai-dev/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  allEnemies.forEach((enemy) => {
+    if (
+      this.x < enemy.x + spriteSize &&
+      this.x + spriteSize > enemy.x &&
+      this.y < enemy.y + spriteSize * 0.75 &&
+      this.y + spriteSize * 0.75 > enemy.y
+    ) {
+      this.x = playerStartX;
+      this.y = playerStartY;
+    }
```
Comment (OleksiyRudenko) [inline submissions/Shrai-dev/frogger-game/app.js:74]: image

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Shrai-dev pretty well done.
Please do not expect mentors to point at exactly every mistake, but rather deduce problematic areas in your code from principles and examples.

**Comment by OleksiyRudenko** [comment]
Does the app work as expected?

**Comment by Shrai-dev** [comment]
Now it works.

**Review by OleksiyRudenko — APPROVED** [review]
@Shrai-dev well done!

Please test your app at all times before requesting review.

**Pull request merged**